src/sentry/integrations/api/serializers/rest_framework/data_forwarder.py

```diff
@@ -0,0 +1,206 @@
+import re
+from collections.abc import MutableMapping
+from typing import Any
+
+from rest_framework import serializers
+from rest_framework.serializers import Serializer, ValidationError
+
+from sentry.integrations.models.data_forwarder import DataForwarder
+from sentry.integrations.models.data_forwarder_project import DataForwarderProject
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+
+
+class DataForwarderSerializer(Serializer):
+    organization_id = serializers.IntegerField()
+    is_enabled = serializers.BooleanField(default=True)
+    enroll_new_projects = serializers.BooleanField(default=False)
+    provider = serializers.ChoiceField(
+        choices=[
+            ("segment", "Segment"),
+            ("sqs", "Amazon SQS"),
+            ("splunk", "Splunk"),
+        ]
+    )
+    config = serializers.JSONField(default=dict)
+
+    def validate_organization_id(self, value: int) -> int:
+        try:
+            Organization.objects.get(id=value)
+        except Organization.DoesNotExist:
+            raise ValidationError("Organization with this ID does not exist")
+        return value
+
+    def validate_config(self, value: dict) -> dict:
+        provider = self.initial_data.get("provider") or getattr(self.instance, "provider", None)
+
+        if provider == "sqs":
+            return self._validate_sqs_config(value)
+        elif provider == "segment":
+            return self._validate_segment_config(value)
+        elif provider == "splunk":
+            return self._validate_splunk_config(value)
+
+        return value
+
+    def _validate_sqs_config(self, config: dict) -> dict:
+        """Validate Amazon SQS specific configuration."""
+        required_fields = ["queue_url", "region", "access_key", "secret_key"]
+        # optional_fields = ["message_group_id", "s3_bucket"]
+
+        missing_fields = [field for field in required_fields if field not in config]
+        if missing_fields:
+            raise ValidationError(f"Missing required SQS fields: {', '.join(missing_fields)}")
+
+        errors = []
+
+        # SQS URL format: https://sqs.region.amazonaws.com/account/queue-name
+        queue_url = config.get("queue_url")
+        sqs_url_pattern = (
+            r"^https://sqs\.[a-z0-9\-]+\.amazonaws\.com/\d+/[a-zA-Z0-9\-_/]+(?:\.fifo)?$"
+        )
+        if not isinstance(queue_url, str) or not re.match(sqs_url_pattern, queue_url):
+            errors.append(
+                "queue_url must be a valid SQS URL format: "
+                "https://sqs.region.amazonaws.com/account/queue-name"
+            )
+
+        aws_region_pattern = r"^[a-z0-9\-]+$"
```

Comment:
Since we're not validating against any regions in particular, it seems like this is a string check? I'm not sure if all the fields we're expecting within the configs are strings but if so we could modify the [serializer field for config](https://www.django-rest-framework.org/api-guide/fields/#dictfield) to specify every nested field must be a String. er lmk if that was confusing